File: app/src/main/java/x/com/dialogdialog/LoginActivity.java
package x.com.dialogdialog;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import x.com.dialogmobile.LoginDialog.LDialogBuilder;

public class LoginActivity extends AppCompatActivity {
    private LDialogBuilder a;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        a = new LDialogBuilder(this).inputphone().setonInputPhoneListener(new LDialogBuilder.onInputPhone() {
            @Override
            public void inputPhonenext(String phone, int type) {
                if (type == 0) {
                    a.dismissdialog1();
                    a.inputcode(phone).setonInputCodeListener(new LDialogBuilder.onInputCode() {
                        @Override
                        public void inputCodenext(String phone, String code) {
                            //校验验证码，接口调用
                        }
                    });
                }
            }
        });
//        a.dismissdialog2();
    }
}

File: app/src/main/java/x/com/dialogdialog/LoginActivity.kt
package x.com.dialogdialog

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import x.com.dialogmobile.LoginDialog.LDialogBuilder

/**
 * 手机号+验证码登录弹窗
 */
class LoginActivity : AppCompatActivity() {
    var a: LDialogBuilder? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        a = LDialogBuilder(this).inputphone()
                .setonInputPhoneListener { phone, i ->
                    //获取验证码，接口调用
                    if (i == 0) {
                        a!!.dismissdialog1()
                        a!!.inputcode(phone).setonInputCodeListener { phone, code ->
                            //校验验证码，接口调用
                        }
                    }
                }

//        a!!.dismissdialog2()


    }
}
